### PR TITLE
CB-74: Add possibility to add photos of recipe

### PR DESCRIPTION
**Pull request by grabomarcin**
https://team-1626248039387.atlassian.net/browse/CB-74

### CODE REVIEW

File: app/models/image.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  include ActiveModel::Validations
+
+  # == Relationships ========================================================
+
+  belongs_to :recipe
+  has_one_attached :photo
+
+  # == Validations ==========================================================
+
+  validates :photo, attached: true, content_type: ['image/png', 'image/jpg', 'image/jpeg']
```
Comment (plokijro) [inline app/models/image.rb:30]: Use constant

**Review by mirecki — APPROVED** [review]

**Review by Narzew — APPROVED** [review]

File: app/controllers/api/v1/images_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
+module Api
+  module V1
+    class ImagesController < ApplicationController
+      before_action :set_recipe
+
+      def create
+        service = ImagesService.new(image_params, @recipe)
```
Comment (h4b00) [inline app/controllers/api/v1/images_controller.rb:7]: ImagesService sounds a bit too general, let's maybe name it `ImagesServiceCreator`
I am a big fan of using kargs if I have more than 1 argument.

File: app/controllers/api/v1/images_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        if service.create
+          render status: :ok
```
Comment (h4b00) [inline app/controllers/api/v1/images_controller.rb:9]:
```suggestion
          render status: :created
```

cuz ok returns `200` but you should return here `201`

File: app/controllers/api/v1/images_controller.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        end
+      end
+
+      def destroy
+        service = ImagesService.new(image_params, @recipe)
+        if service.destroy
+          render status: :ok
+        else
+          render json: { errors: service.errors.full_messages }, status: :unprocessable_entity
+        end
```
Comment (h4b00) [inline app/controllers/api/v1/images_controller.rb:21]: Do you have a CRUD action for destroying multiple resources?

File: app/services/images_service.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    images = @image_params[:images]
+
+    return unless @recipe.validate_number_of_images(images.count)
+
+    ActiveRecord::Base.transaction do
+      images.each do |image|
+        new_image = @recipe.images.new
+        new_image.photo.attach(image)
+        new_image.save!
+      end
```
Comment (h4b00) [inline app/services/images_service.rb:20]: That sounds strange why do we want to upload multiple files like this at once?
I mean every time when you guys upload multiple files, In the background you are doing this one by one.
So you don't have a problem with upload file size limit

**Review by h4b00 — CHANGES_REQUESTED** [review]
Why swagger.json is much smaller than before?

File: app/controllers/api/v1/images_controller.rb (hunk 4 above)
Comment (grabomarcin) [reply app/controllers/api/v1/images_controller.rb:21]: I  did it in Image service, it gets an array of id's and destroys them.

File: app/services/images_service.rb (hunk 5 above)
Comment (grabomarcin) [reply app/services/images_service.rb:20]: Would it be a better solution if FE hit several times in end-point sending photos one by one?

File: app/controllers/api/v1/images_controller.rb (hunk 2 above)
Comment (grabomarcin) [reply app/controllers/api/v1/images_controller.rb]: Good point. I changed that

File: app/controllers/api/v1/images_controller.rb (hunk 3 above)
Comment (grabomarcin) [reply app/controllers/api/v1/images_controller.rb]: Fixed.

**Pull request closed without merging**